gax/src/main/java/com/google/api/gax/batching/BatchingSettings.java

```diff
@@ -174,6 +174,21 @@ public BatchingSettings build() {
           settings.getDelayThreshold() == null
               || settings.getDelayThreshold().compareTo(Duration.ZERO) > 0,
           "delayThreshold must be either unset or positive");
+      if (settings.getFlowControlSettings().getLimitExceededBehavior()
+          != LimitExceededBehavior.Ignore) {
+        Preconditions.checkArgument(
+            settings.getFlowControlSettings().getMaxOutstandingElementCount() == null
+                || settings.getElementCountThreshold() == null
+                || settings.getFlowControlSettings().getMaxOutstandingElementCount()
+                    >= settings.getElementCountThreshold(),
+            "if throttling and batching on element count are enabled, FlowController#maxOutstandingElementCount must be greater or equal to elementCountThreshold");
+        Preconditions.checkArgument(
+            settings.getFlowControlSettings().getMaxOutstandingRequestBytes() == null
+                || settings.getRequestByteThreshold() == null
+                || settings.getFlowControlSettings().getMaxOutstandingRequestBytes()
+                    >= settings.getRequestByteThreshold(),
+            "if throttling and batching on request bytes are enabled, FlowController#maxOutstandingRequestBytes must be greater or equal to requestByteThreshold");
```

Comment:
Should those error messages start with capital letter? Also, please split the error sting into mutiple lines, as this one is 160+ characters long and exceeds the formatting limits.